types/node/index.d.ts

```diff
@@ -748,6 +748,11 @@ declare module "http" {
     export interface ServerRequest extends IncomingMessage {
         connection: net.Socket;
     }
+
+    export interface ServerResponseHeaders {
+      [key: string]: number | string | string[];
```

Comment:
I have added number to getHeader. For all normal use cases people don't have to change anything in their code. If they really want to do strange/useless things like adding objects or mixed arrays to header variables then they can cast to make it possible.